addons/stock/models/product.py

```diff
@@ -132,17 +132,19 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
 
         res = dict()
         for product in self.with_context(prefetch_fields=False):
-            res[product.id] = {}
+            product_id = product.id
+            rounding = product.product_tmpl_id.uom_id.rounding
```

Comment:
@nim-odoo l'update pour la performance est surtout ici.

Reply:
For this, we would need a script to reproduce the issue, since it seems similar to #24483. This could be a more generic problem to solve.